openshift2nulecule/openshift.py

```diff
@@ -112,19 +112,25 @@ def export_project(self):
         # Exporting ReplicationControllers should be enough.
         # Ideally this should detect Pods that are not created by
         # ReplicationController and only export those.
+        # Order in resource list is significant! Object are exported in same
+        # order as they are specified on command line and they will have same
+        # order in Nulecule file also.
+        # ImageStream is first as workaround for this https://github.com/openshift/origin/issues/4518
+        # But this workaround is going to work only after resolving
+        # https://github.com/projectatomic/atomicapp/issues/669
         all_artifacts = {}
         for provider in NULECULE_PROVIDERS:
             if provider == "kubernetes":
-                resources = ["replicationController",
-                             "persistentVolumeClaim",
-                             "service"]
+                resources = ["persistentVolumeClaim",
+                             "service",
+                             "replicationController"]
```

Comment:
They won't have the same order in Nulecule file, they will have order in which [`os.listdir(nulecule_dir)`](https://github.com/projectatomic/openshift2nulecule/blob/master/openshift2nulecule/utils.py#L79) lists the files in `utils.generate_dockerfile`.

Again we cannot be so sure, how `Atomic App` is passing it out to respective providers. But sure we can make sequence in Nulecule file, and take care of `Atomic App` side of things.